qiskit/pulse/parameter_manager.py

```diff
@@ -415,7 +416,7 @@ def _unroll_param_dict(
                 if not isinstance(value, Sequence):
                     raise PulseError(
                         f"Parameter vector '{parameter.name}' has length {len(parameter)},"
-                        f" but was assigned to a single value."
+                        f" but was assigned to {value}."
```

Comment:
Overall, the Pulse model of assigning parameters was very poor in checking the variables input (it was left to the class `ParameterExpression`during the binding. We could keep going this track and make the type checking, as well as the input type as minimal as possible (hence removing those warnings), to facilitate a further Jax compatible workflow (where such checks would have to be disabled to integrate the parameter assigmnent within a Jit compiled function. Another alternative for this last aspect is to add a class attribute disable_validation (similarly to what was proposed in #11029) to the `Parameter` class directly (to avoid any additional Python check on traced variables).

Reply:
That is a reasonable path. My comment here was just about whether `value` should be printed if the lengths don't match below as well. Maybe we can just leave it printing the lengths for now.